### ISSUE TITLE
Add WAF extension

### ISSUE THREAD

**Issue by bianpengyuan** [istio-ecosystem/wasm-extensions#13]
This is the track the effort to get https://github.com/istio-ecosystem/wasm-extensions/pull/2 in. The remaining work includes:
* Add unit test and integration test
* Use bazel for its build
* Add doc about how to customize the extension

cc @fpesce I will add guide on how to use bazel and add tests coverage to a Wasm extension.

**Comment by verysonglaa** [comment]
Has there been any progress lately?

**Comment by bianpengyuan** [comment]
> Has there been any progress lately?

Nope, but it would be great if someone could pick up https://github.com/istio-ecosystem/wasm-extensions/pull/2 and update it to a mergeable state.
